internal/querynodev2/delegator/delegator_data.go

```diff
@@ -668,6 +668,9 @@ func (sd *shardDelegator) loadStreamDelete(ctx context.Context,
 			}
 		}
 
+		// if segment level is not L0, delete buffer could be purged safely according to target checkpoint
+		sd.deleteBuffer.TryDiscard(position.GetTimestamp())
```

Comment:
should we change the synctarget version func?

instead use checkpoint timestamp, we can use L0 segment's last message?